src/hardware/setup.rs

```diff
@@ -724,6 +724,7 @@ pub fn setup(
     fp_led_2.set_low();
     fp_led_3.set_low();
 
+    // Use default PLL2p clock input with 1/2 prescaler for 50 MHz ADC clock.
```

Comment:
I think that PLL2p = 100MHz and PRESC = div 2 actually gives 25MHz a ADC clock on current parts. There's an extra /2 at the multiplexer output in RM0433 Figure 136.

![image](https://user-images.githubusercontent.com/962920/184555700-6143484a-66ff-4185-bfe7-0a0fab8b5f95.png)

I assume this ADC clock after the /2 is the clock denoted f_ADC in the datasheet ![image](https://user-images.githubusercontent.com/962920/184555783-6f97b6d6-2a94-4a02-a4db-17ae1a7a4142.png)

Reply:
Thanks. I would think you're right. They just drop the "ker" later and call this f_ADC. 
The only thing is that we consistently observed that the ADC reading are off using a 100 MHz clock input without an additional divider (aka this should make f_ADC = 50 MHz). This should be allowed with BOOST = 11 according to the datasheet. This was independent of the sample time or any other ADC variables AFAICT.

Reply:
I find RM0433 kinda confusing regarding the ADC clocking...
![Screenshot from 2022-08-15 10-13-44](https://user-images.githubusercontent.com/17088194/184600501-bc6221a9-cfd7-48bf-84e0-f1642916d027.png)
How am I supposed to understand this? I assume  they mean the F_adc_ker_ck_input. Or is there another /2 divider behind F_adc_ker_ck?

Reply:
50 MHz is incompatible with the 80 MHz value for adc_ker_ck_input in RM0433, which is definitely before CCR:PREC and thus before the div2 no matter how one looks at it (and no matter whether the 50 MHz are before or after the div2.

![image](https://user-images.githubusercontent.com/1338946/184605985-b1c9d185-055d-44fc-b69d-837c771d0ee7.png)

Since the 80 MHz adc_ker_ck_input is the tighter constraint and matches the dramatic drop of between 80 MHz and 100 MHz we have seen, I'd go for that and enforce max 40 MHz adc_ker_ck.
Note the requirement for 50% duty and even PLL divider which might be a bit hairier to enforce. (Also note the typo VOS2/VOS3 in the column header and that I find it unlikely that VOS2 doesn't have this requirement).